File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/data_processing_template.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-


#Data Processing

#Import Libararies
import numpy as np #mathematical tools
import matplotlib.pyplot as plt #for ploting charts
import pandas as pd #import data sets and manage data sets

dataset = pd.read_csv('Data.csv')
X = dataset.iloc[:, :-1].values #: - takes all the values in row and :- take all the column except the last one
Y = dataset.iloc[:, 3].values
                
#Taking care of missing data
from sklearn.preprocessing import Imputer

imputer = Imputer(missing_values= "NaN", strategy= "mean", axis=0) #strategy default value is mean, axis = 1 => row, axis = 0 => column
imputer = imputer.fit(X[:,1:3]) #The upper bound 3 is excluded
X[:,1:3] = imputer.transform(X[:,1:3])

File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/data_processing_template.R
dataset = read.csv('Data.csv') #you dont have to distinguish between dependent and independent variables

#Taking care of missing data
dataset$Age = ifelse(is.na(dataset$Age), ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)), dataset$Age) #na.rm = true to calculate mean for the non missing data
dataset$Salary = ifelse(is.na(dataset$Salary), ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)), dataset$Salary)
                                                                                              
